virt-do/cloudlet
# feat(vmm): add grpc server to handle api requests

### What does this pr do:

This pr adds a gRPC server to our vmm to handle run requests from the api.


### How can this be tested: 

- run the api using `cargo run --bin api`
- build and run the vmm using 
```bash 
cargo build
sudo -E capsh --keep=1 --user=$USER --inh=cap_net_admin --addamb=cap_net_admin -- -c  'RUST_BACKTRACE=1 '$CARGO_PATH' run --bin vmm -- grpc'
```
- fill the yaml template in src/cli/config/config.template.yaml and make sure the code_path is pointing to a real file
- run the cli and send a query 
```bash
cargo run --bin cli run --config-path <path_to_cloudlet>/src/cli/config/config.yaml
```


You should see a new vm get runned in the vmm shell you started
